ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/numberinput/v1/numberinput/clientlibs/editor/js/editDialog.js

```diff
@@ -52,6 +52,37 @@
     }
      **/
 
+    function handleTypeDropdown(dialog) {
+        let typeDropdownComponent = dialog.find(NUMERICINPUT_TYPE)[0];
+        manageMinAndMaxValidation();
+        typeDropdownComponent.addEventListener("change", manageMinAndMaxValidation );
+
+        function manageMinAndMaxValidation() {
+            let minField = dialog.find(NUMERICINPUT_MINIMUM)[0];
+            let maxField = dialog.find(NUMERICINPUT_MAXIMUM)[0];
+            let minInput = minField.querySelector('input');
+            let maxInput = maxField.querySelector('input');
+            // update step property to 0.01 for minField and maxField
+            if(typeDropdownComponent.value === 'number'){
+                minField.step = "0.01";
+                maxField.step = "0.01";
+            } else {
+                minField.step = "1";
+                maxField.step = "1";
+                // Reset value if it is a decimal
+                if (minInput.value.includes('.')) {
+                    minInput.value = Math.floor(minInput.value);
+                    minInput.setAttribute('value', minInput.value);
+                }
+                if (maxInput.value.includes('.')) {
+                    maxInput.value = Math.floor(maxInput.value);
+                    maxInput.setAttribute('value', maxInput.value);
+                }
```

Comment:
can we add a test case for this use case where changing from number to integer, changes the min/max constraint as well